### PR TITLE
Enhance GKE DRANET support for multi-node pool configuration 

### PR DESCRIPTION
**Pull request by agrawalkhushi18**
This PR enhances GKE DRANET support for clusters with multi-node pool configurations. It prevents Helm resource duplication conflicts during deployment and dynamically configures workload network claims based on the machine type (RDMA or non-RDMA supported) .

## Summary of Changes
*   **`pkg/config/expand.go`**: Implements Go compiler logic to detect and deduplicate DRANET Helm templates across multiple GKE node pools sharing the same network device class. Added query parameter stripping to ensure remote modules are correctly identified in production.
*   **`pkg/config/expand_test.go`**: Adds comprehensive unit tests validating the compiler's deduplication under various scenarios (same class, mixed GPU/TPU pools, custom overrides, and remote module URL structures).
*   **`modules/compute/gke-job-template` (`main.tf`, `variables.tf`)**: Updates the workload template to dynamically request the correct template name (`default-netdev` vs `default-rdma`) from the node pool, adds null-safety for mixed configurations, and renames the local claim alias to a generic name (`dranet-network`).
*   **`modules/compute/gke-node-pool` (`main.tf`, `variables.tf`, `outputs.tf`)**: Exposes the resolved DRANET template name via outputs so the job template can inherit it, and documents the compiler's auto-deduplication behavior in the variables.

### 1. Unit Tests
*   Ran all compiler unit tests using `make test-engine` (all passed, including the new edge-case tests for deduplication and remote module paths).

### 2. E2E Cluster Deployment & Workload Run
*   Successfully compiled and deployed a GKE cluster with 2 TPU v6e node pools (topology `4x4`, 32 chips total) in `southamerica-west1` with DRANET enabled.
*   Successfully ran a multi-slice JAX initialization workload (`./gcluster job submit`) across both slices (32 chips total). The JAX coordination service initialized successfully on all 8 pods without hangs, confirming multi-NIC policy routing is functional.

### Submission Checklist

NOTE: Community submissions can take up to 2 weeks to be reviewed.

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cluster Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: pkg/config/expand.go [hunk 1]
```diff
@@ -575,3 +578,75 @@ func intersection(s1 []string, s2 []string) []string {
...
+		}
+
+		if !bp.getEnableDranet(m) {
+			return
+		}
+
+		deviceClassName := bp.getDeviceClassName(m)
+
+		// Check for duplicates
+		if primaryModuleID, exists := installedTemplates[deviceClassName]; exists {
```
[the remark this answers is not in the document]
Comment (agrawalkhushi18) [reply pkg/config/expand.go:643]: NA

File: modules/compute/gke-job-template/main.tf [hunk 2]
```diff
@@ -125,7 +125,10 @@ locals {
     pvc.storage_type == "gcs"
   ])
 
-  default_resource_claims = anytrue(var.enable_dranet) ? [{ name = "rdma", resource_claim_template_name = "default-rdma" }] : []
+  dranet_template_name_distinct = distinct(var.dranet_template_name)
+  dranet_template_name_actual   = length(local.dranet_template_name_distinct) > 0 ? local.dranet_template_name_distinct[0] : "default-rdma"
```
[the remark this answers is not in the document]
Comment (agrawalkhushi18) [reply modules/compute/gke-job-template/main.tf]: done

File: pkg/config/expand.go [hunk 3]
```diff
@@ -575,3 +578,75 @@ func intersection(s1 []string, s2 []string) []string {
 	slices.Sort(is)
 	return is
 }
+
+func isGKENodePool(m *Module) bool {
+	return strings.HasSuffix(m.Source, "modules/compute/gke-node-pool")
+}
```
[the remark this answers is not in the document]
Comment (agrawalkhushi18) [reply pkg/config/expand.go:585]: done

File: pkg/config/expand_test.go [hunk 4]
```diff
@@ -589,3 +589,123 @@ func (s *zeroSuite) TestCheckInputValueMatchesType_Panic(c *C) {
...
+			}),
+		}
+		m2 := Module{
+			ID:     "pool-b",
+			Source: "community/modules/compute/gke-node-pool",
+			Settings: NewDict(map[string]cty.Value{
+				"enable_dranet": cty.BoolVal(true),
+				"machine_type":  cty.StringVal("ct6e-standard-4t"), // auto-detects to netdev.google.com
+			}),
+		}
```
[the remark this answers is not in the document]
Comment (agrawalkhushi18) [reply pkg/config/expand_test.go:612]: Added new test.

File: pkg/config/expand.go [hunk 5]
```diff
@@ -32,7 +33,7 @@ import (
 const (
 	blueprintLabel        = "ghpc_blueprint"
 	deploymentLabel       = "ghpc_deployment"
-	GoogleProviderVersion = ">= 6.9.0, <= 7.38.0"
+	GoogleProviderVersion = ">= 6.9.0, <= 7.36.0"
```
Comment (Neelabh94) [inline pkg/config/expand.go]: please rebase with the latest develop.

Comment (agrawalkhushi18) [reply pkg/config/expand.go]: Done, Thanks!

**Review by SwarnaBharathiMantena — APPROVED** [review]
LGTM
